src/shared/components/case-editor/services/wizard-page-field-to-case-field.mapper.ts

```diff
@@ -0,0 +1,84 @@
+import { WizardPageField } from '../domain';
+import { CaseField } from '../../../domain';
+import { ComplexFieldMask } from '../domain/wizard-page-field-complex-mask.model';
+import { Injectable } from '@angular/core';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class WizardPageFieldToCaseFieldMapper {
+
+  mapAll(wizardPageField: WizardPageField[], caseFields: CaseField[]): CaseField[] {
+    return wizardPageField.map((wizardField: WizardPageField) => {
+      return this.map(wizardField, caseFields);
+    });
+  }
+
+  private map(wizardPageField: WizardPageField, caseFields: CaseField[]): CaseField {
+
+    let case_field: CaseField = caseFields.find(e => e.id === wizardPageField.case_field_id);
```

Comment:
starting to think it would be better to organise the caseFields in a Map. Even thought this might be a bit too much of a change now. Created a separate ticket https://tools.hmcts.net/jira/browse/RDM-4054